extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -830,6 +830,13 @@ export class AmpA4A extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    // Shorthand for: reject promise if current promise chain is out of date.
+    const checkStillCurrent = promiseId => {
+      if (promiseId != this.promiseId_) {
+        throw cancellation();
+      }
+    };
+    const promiseId = this.promiseId_;
```

Comment:
I would do:
    `const promiseId = this.promiseId_;`
     `const checkStillCurrent = () => {`
     `  if (promiseId != this.promiseId_) {`
     `    throw cancellation();`
     `  }`
   `};`
Then you shouldn't need to pass down promiseId